### ISSUE TITLE
Kubernetes: worker pod consumes tasks while terminating

### ISSUE THREAD

**Issue by callumforrester** [DiamondLightSource/blueapi#95]
When blueapi is running as a pod, and it is upgraded or restarted, it will be set to terminating status and a new pod will be spun up before it is gone. This can be a problem as the soon-to-be-terminated pod can continue to consume tasks from the queue, effectively stealing them from the new pod.

**Comment by callumforrester** [comment]
May be superseded by #120

**Comment by DiamondJoseph** [comment]
Still an issue after #120 as tasks now in the service's queue

**Comment by callumforrester** [comment]
Except the task queue as a whole will be redundant after #120

**Comment by callumforrester** [comment]
Non-issue after #120
